packages/backend/src/indexers/CacheInvalidationIndexer.ts

```diff
@@ -0,0 +1,64 @@
+import { assert, Logger } from '@l2beat/backend-tools'
+import { ChildIndexer } from '@l2beat/uif'
+import { ChainId, UnixTime } from '@lz/libs'
+
+import { BlockNumberRepository } from '../peripherals/database/BlockNumberRepository'
+import { IndexerStateRepository } from '../peripherals/database/IndexerStateRepository'
+import { ProviderCacheRepository } from '../peripherals/database/ProviderCacheRepository'
+import { BlockNumberIndexer } from './BlockNumberIndexer'
+
+export class CacheInvalidationIndexer extends ChildIndexer {
+  private readonly id = 'CacheInvalidationIndexer'
+  constructor(
+    private readonly blockNumberRepository: BlockNumberRepository,
+    private readonly cacheRepository: ProviderCacheRepository,
+    private readonly indexerStateRepository: IndexerStateRepository,
+    private readonly chainId: ChainId,
+    logger: Logger,
+    blockNumberIndexer: BlockNumberIndexer,
+  ) {
+    super(logger.tag(ChainId.getName(chainId)), [blockNumberIndexer])
+  }
+
+  async update(_fromTimestamp: number, toTimestamp: number): Promise<number> {
+    // NOOP
+    return Promise.resolve(toTimestamp)
+  }
+
+  override async invalidate(targetHeight: number): Promise<number> {
+    if (targetHeight === 0) {
+      return targetHeight
+    }
```

Comment:
There at least should be a log explaining why we skip invalidation.

Reply:
https://github.com/l2beat/lz-monitoring/pull/37/commits/66432c48b3c717441150dfe655db03a98476d4ed